### PR TITLE
Added support for custom menu node registration.

### PR DESCRIPTION
**Pull request by kittaakos**
Signed-off-by: Akos Kitta <kittaakos@typefox.io>

#### What it does

From now on, it's possible to register a custom menu node into the menu registry.

Added a placeholder example:
Screen Shot 2020-08-19 at 13 38 14

Screen Shot 2020-08-19 at 13 37 26

#### How to test

Check the sample apps.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by akosyakov** [comment]
I will review API changes tomorrow morning.

**Comment by kittaakos** [comment]
> I will review API changes tomorrow morning.

Sure. No hurry at all. It is not urgent for me. It can even wait until next week if you're busy.

File: packages/core/src/browser/menu/browser-menu-plugin.ts [hunk 1]
```diff
@@ -99,22 +103,30 @@ export class BrowserMainMenuFactory {
...
+    }
+
+    protected get services(): MenuServices {
         return {
             context: this.context,
             contextKeyService: this.contextKeyService,
             commandRegistry: this.commandRegistry,
-            keybindingRegistry: this.keybindingRegistry
+            keybindingRegistry: this.keybindingRegistry,
+            meneWidgetFactory: this
```
Comment (akosyakov) [inline packages/core/src/browser/menu/browser-menu-plugin.ts:123]: `menu`

**Review by akosyakov — APPROVED** [review]
LGTM

File: packages/core/src/browser/menu/browser-menu-plugin.ts (hunk 1 above)
Comment (kittaakos) [reply packages/core/src/browser/menu/browser-menu-plugin.ts:123]: Ohh, thank you!

**Pull request merged**